src/segmentTemplate.js

```diff
@@ -0,0 +1,277 @@
+import { range } from './utils/list';
+import resolveUrl from './resolveUrl';
+
+const identifierPattern = /\$([A-z]*)(?:(%0)([0-9]+)d)?\$/g;
+
+/**
+ * Returns a function to be used as a callback for String.prototype.replace to replace
+ * template identifiers
+ *
+ * @param {Obect} values
+ *        Object containing values that shall be used to replace known identifiers
+ * @param {number} values.RepresentationID
+ *        Value of the Representation@id attribute
+ * @param {number} values.Number
+ *        Number of the corresponding segment
+ * @param {number} values.Bandwidth
+ *        Value of the Representation@bandwidth attribute.
+ * @param {number} values.Time
+ *        Timestamp value of the corresponding segment
+ * @return {(match: string, identifier: string, format: string, width: string) => string}
```

Comment:
Following this format https://github.com/jsdoc3/jsdoc/issues/1286#issuecomment-341908946

Our linter isn't too happy with it `warning  JSDoc syntax error         valid-jsdoc`. Not sure of a better way of expressing this. It won't prevent builds from passing at least